lib/CsvImportValidator.class.php

```diff
@@ -0,0 +1,466 @@
+ */
+
+require_once __DIR__.'/../vendor/composer/autoload.php';
+
+/**
+ * Check csv data.
+ *
+ * @author     Steve Breker <sbreker@artefactual.com>
+ */
+class CsvImportValidator
+{
+    const UTF8_BOM = "\xEF\xBB\xBF";
+    const UTF16_LITTLE_ENDIAN_BOM = "\xFF\xFE";
+    const UTF16_BIG_ENDIAN_BOM = "\xFE\xFF";
+    const UTF32_LITTLE_ENDIAN_BOM = "\xFF\xFE\x00\x00";
+    const UTF32_BIG_ENDIAN_BOM = "\x00\x00\xFE\xFF";
+
+    public static $bomTypeMap = [
+        'utf8Bom' => self::UTF8_BOM,
+        'utf16LittleEndianBom' => self::UTF16_LITTLE_ENDIAN_BOM,
+        'utf16BigEndianBom' => self::UTF16_BIG_ENDIAN_BOM,
+        'utf32LittleEndianBom' => self::UTF32_LITTLE_ENDIAN_BOM,
+        'utf32BigEndianBom' => self::UTF32_BIG_ENDIAN_BOM,
+    ];
+    protected $context;
+    protected $dbcon;
+    protected $filenames = [];
+    protected $csvTests;
+    protected $header;
+    protected $rows = [];
+    protected $showDisplayProgress = false;
+    protected $resultCollection;
+    protected $ormClasses = [];
+
+    // Default options:
+    // Assumes QubitInformationObject if className option not set.
+    protected $validatorOptions = [
+        'className' => 'QubitInformationObject',
+        'verbose' => false,
+        'source' => '',
+        'separator' => ',',
+        'enclosure' => '"',
+        'specificTests' => '',
+        'pathToDigitalObjects' => '',
+    ];
+
+    // List of valid classNames
+    protected $defaultCsvClassNameList = [
+        'QubitInformationObject',
+        'QubitActor',
+        'QubitAccession',
+        'QubitRepository',
+        'QubitEvent',
+    ];
+
+    // General tests which can apply to any CSV.
+    protected $generalTestList = [
+        'CsvSampleValuesValidator' => CsvSampleValuesValidator::class,
+        'CsvFileEncodingValidator' => CsvFileEncodingValidator::class,
+        'CsvColumnNameValidator' => CsvColumnNameValidator::class,
+        'CsvColumnCountValidator' => CsvColumnCountValidator::class,
+        'CsvDuplicateColumnNameValidator' => CsvDuplicateColumnNameValidator::class,
+        'CsvEmptyRowValidator' => CsvEmptyRowValidator::class,
+        'CsvCultureValidator' => CsvCultureValidator::class,
+        'CsvLanguageValidator' => CsvLanguageValidator::class,
+        'CsvFieldLengthValidator' => CsvFieldLengthValidator::class,
+    ];
+
+    // Tests which pertain only to this class type.
+    protected $qubitInformationObjectTestList = [
+        'CsvParentValidator' => CsvParentValidator::class,
+        'CsvLegacyIdValidator' => CsvLegacyIdValidator::class,
+        'CsvEventValuesValidator' => CsvEventValuesValidator::class,
+        'CsvDigitalObjectPathValidator' => CsvDigitalObjectPathValidator::class,
+        'CsvDigitalObjectUriValidator' => CsvDigitalObjectUriValidator::class,
+    ];
+
+    public function __construct(sfContext $context = null,
+    $dbcon = null,
+    $options = [])
```

Comment:
Cosmetic: Closing parenthesis should be on next line

Reply:
fixed